File: source/python/markdown_to_html.py
import re

def convert_inline(line):
    result = line
    # Bold: **text**
    while "**" in result:
        start = result.find("**")
        rest = result[start + 2:]
        end = rest.find("**")
        if end < 0:
            break
        bold_text = rest[:end]
        before = result[:start]
        after = rest[end + 2:]
        result = f"{before}<strong>{bold_text}</strong>{after}"

    # Italic: *text*
    while "*" in result:
        start = result.find("*")
        rest = result[start + 1:]
        end = rest.find("*")
        if end < 0:
            break
        ital_text = rest[:end]
        before = result[:start]
        after = rest[end + 1:]
        result = f"{before}<em>{ital_text}</em>{after}"

    # Inline code: `text`
    while "`" in result:
        start = result.find("`")
        rest = result[start + 1:]
        end = rest.find("`")
        if end < 0:
            break
        code_text = rest[:end]
        before = result[:start]
        after = rest[end + 1:]
        result = f"{before}<code>{code_text}</code>{after}"

    return result

def md_to_html(text):
    lines = text.split("\n")
    html = []
    for line in lines:
        if line.startswith("### "):
            html.append(f"<h3>{convert_inline(line[4:])}</h3>")
        elif line.startswith("## "):
            html.append(f"<h2>{convert_inline(line[3:])}</h2>")
        elif line.startswith("# "):
            html.append(f"<h1>{convert_inline(line[2:])}</h1>")
        elif line.startswith("- "):
            html.append(f"<li>{convert_inline(line[2:])}</li>")
        elif line.strip() == "":
            html.append("")
        else:
            html.append(f"<p>{convert_inline(line)}</p>")
    return "\n".join(html)

markdown = "# Hello World\n\nThis is **bold** and *italic* text.\n\n## Features\n\n- Item with `code`\n- Another **bold** item\n\n### Details\n\nA paragraph with **mixed** and *styles*."

html = md_to_html(markdown)
print(html)

File: source/javascript/markdown_to_html.js
// Basic markdown to HTML converter
function convertInline(line) {
    let result = line;

    // Bold: **text**
    while (result.includes("**")) {
        const start = result.indexOf("**");
        const rest = result.slice(start + 2);
        const end = rest.indexOf("**");
        if (end < 0) break;
        const boldText = rest.substring(0, end);
        const before = result.substring(0, start);
        const after = rest.slice(end + 2);
        result = `${before}<strong>${boldText}</strong>${after}`;
    }

    // Italic: *text*
    while (result.includes("*")) {
        const start = result.indexOf("*");
        const rest = result.slice(start + 1);
        const end = rest.indexOf("*");
        if (end < 0) break;
        const italText = rest.substring(0, end);
        const before = result.substring(0, start);
        const after = rest.slice(end + 1);
        result = `${before}<em>${italText}</em>${after}`;
    }

    // Inline code: `text`
    while (result.includes("`")) {
        const start = result.indexOf("`");
        const rest = result.slice(start + 1);
        const end = rest.indexOf("`");
        if (end < 0) break;
        const codeText = rest.substring(0, end);
        const before = result.substring(0, start);
        const after = rest.slice(end + 1);
        result = `${before}<code>${codeText}</code>${after}`;
    }

    return result;
}

function mdToHtml(text) {
    const lines = text.split("\n");
    const html = [];
    for (const line of lines) {
        if (line.startsWith("### ")) html.push(`<h3>${convertInline(line.slice(4))}</h3>`);
        else if (line.startsWith("## ")) html.push(`<h2>${convertInline(line.slice(3))}</h2>`);
        else if (line.startsWith("# ")) html.push(`<h1>${convertInline(line.slice(2))}</h1>`);
        else if (line.startsWith("- ")) html.push(`<li>${convertInline(line.slice(2))}</li>`);
        else if (line.trim() === "") html.push("");
        else html.push(`<p>${convertInline(line)}</p>`);
    }
    return html.join("\n");
}

const markdown = "# Hello World\n\nThis is **bold** and *italic* text.\n\n## Features\n\n- Item with `code`\n- Another **bold** item\n\n### Details\n\nA paragraph with **mixed** and *styles*.";
const html = mdToHtml(markdown);
console.log(html);
